src/panels/config/integrations/integration-panels/mqtt/mqtt-config-panel.ts

```diff
@@ -54,6 +60,18 @@ class HaPanelDevMqtt extends LitElement {
                 .value=${this.topic}
                 @change=${this._handleTopic}
               ></ha-textfield>
+              <ha-select
+                .label=${this.hass.localize("ui.panel.config.mqtt.qos")}
+                .value=${this.qos}
+                @selected=${this._handleQos}
+                @closed=${stopPropagation}
+                fixedMenuPosition
+                naturalMenuWidth
+                >${["0", "1", "2"].map(
```

Comment:
make this array a `const` outside the class, so it is not recreated on every render.